analyzers/src/SonarAnalyzer.CSharp/Rules/GetTypeWithIsAssignableFromCodeFixProvider.cs

```diff
@@ -29,143 +29,141 @@
+                _ => null
+            };
 
-            return TaskHelper.CompletedTask;
+        private static SyntaxNode ChangeInvocation(SyntaxNode root, Diagnostic diagnostic, InvocationExpressionSyntax invocation)
+        {
+            var useIsOperator = bool.Parse(diagnostic.Properties[GetTypeWithIsAssignableFrom.UseIsOperatorKey]);
+            var shouldRemoveGetType = bool.Parse(diagnostic.Properties[GetTypeWithIsAssignableFrom.ShouldRemoveGetTypeKey]);
+            var newNode = RefactoredExpression(invocation, useIsOperator, shouldRemoveGetType);
+            return root.ReplaceNode(invocation, newNode.WithAdditionalAnnotations(Formatter.Annotation));
         }
 
-        private static bool TryGetNewRoot(SyntaxNode root, Diagnostic diagnostic, InvocationExpressionSyntax invocation,
-            BinaryExpressionSyntax binary, out SyntaxNode newRoot)
+        private static SyntaxNode ChangeBinary(SyntaxNode root, BinaryExpressionSyntax binary)
         {
-            newRoot = null;
-            if (invocation != null)
+            if (binary.IsKind(SyntaxKind.IsExpression))
             {
-                newRoot = ChangeInvocation(root, diagnostic, invocation);
+                return ChangeIsExpressionToNullCheck(root, binary);
             }
-            else if (binary.IsKind(SyntaxKind.IsExpression))
+            else if (RefactoredExpression(binary) is { } expression)
             {
-                newRoot = ChangeIsExpressionToNullCheck(root, binary);
+                return root.ReplaceNode(binary, expression.WithAdditionalAnnotations(Formatter.Annotation));
             }
             else
             {
-                if (TryGetRefactoredExpression(binary, out var expression))
-                {
-                    newRoot = root.ReplaceNode(binary, expression.WithAdditionalAnnotations(Formatter.Annotation));
-                }
+                return null;
             }
-
-            return newRoot != null;
         }
 
-        private static SyntaxNode ChangeIsExpressionToNullCheck(SyntaxNode root, BinaryExpressionSyntax binary)
+        private static SyntaxNode ChangeIsPattern(SyntaxNode root, IsPatternExpressionSyntaxWrapper isPattern)
         {
-            var newNode = GetExpressionWithParensIfNeeded(GetNullCheck(binary), binary.Parent);
-            return root.ReplaceNode(binary, newNode.WithAdditionalAnnotations(Formatter.Annotation));
+            if (isPattern.Expression is BinaryExpressionSyntax binary)
+            {
+                var negationRequired = true;
+                var current = isPattern.Pattern;
+                while (current.SyntaxNode.IsKind(SyntaxKindEx.NotPattern))
+                {
+                    negationRequired = !negationRequired;
+                    current = ((UnaryPatternSyntaxWrapper)current).Pattern;
+                }
+                var newExpression = NegatedExpression(negationRequired, isPattern.SyntaxNode.Parent, GetIsExpression(binary));
```

Comment:
this should not reuse the same `NegatedExpression()` for `IsPattern` , because `!(x is Y)` should actually be `x is not Y`

Reply:
I opened https://github.com/SonarSource/sonar-dotnet/issues/4633 for this.